swift/WorkflowUI/Sources/Screen/ViewEnvironment/EnvironmentScreen.swift

```diff
@@ -0,0 +1,57 @@
+/*
+* Copyright 2012 Square Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+/// A wrapper screen that overrides environment values
+public struct EnvironmentScreen<Value, Content> {
```

Comment:
Is there value in this? Without this `ViewEnvironment` is fully contained within `viewControllerDescription` and `UIViewController`s.

Reply:
`ViewControllerDescription` itself doesn’t have any knowledge of `ViewEnvironment`.

My desire for this is for wrapper workflows to be able to provide environment values without having to implement a wrapper screen themselves. Let’s use automatic day/night theme switching as an example (ignoring iOS’s dark mode support for a moment). I could have a workflow that has itself subscribes to some signal for time of day, changing its state to `.day` or `.night` as appropriate. With this extension, it’s `render` could be:
```
func render(state: State, context: RenderContext<AutoNightModeWorkflow>) -> Rendering {
    return wrappedWorkflow
        .mapRendering { $0.environment(\.theme, Theme(style: state.timeOfDay) }
        .rendered(with: context)
}
```

Reply:
Doesn't this break the UI/Workflow decoupling? `ViewEnvironment` is purely a view concern, so the workflows shouldn't know about it.

Reply:
IMO, it’s okay for a workflow to put values _into_ the environment. But not to take them out. We absolutely could make a workflow like the one described above do the same job by implementing a wrapper screen type, but it’s effectively going to do the same thing and be the same flow of information.

Reply:
Can we have this discussion in a separate PR, please? It's a big change from the Kotlin behavior, for a speculative use case. And it's not required by the general machinery here, right?

Reply:
Sure. I’ll kick it out of this PR when I update to fix the tutorial code.

Reply:
While I'm thinking about it, though: one thing to note in defense of this class is that container views themselves can transform child renderings into other screen classes. I've already done that a few times, very handy.

That said, I still don't think this class should be a part of this PR.

Reply:
Actually, thinking more about this - this might be the primary way for us to set environment values.

Reply:
https://github.com/square/workflow/issues/1001